usaspending_api/awards/tests/test_models.py

```diff
@@ -0,0 +1,138 @@
+        'awards.Transaction',
+        award=award,
+        awarding_agency=agency2,
+        period_of_performance_current_end_date=datetime.date(2017, 1, 1),
+        description='new description',
+        action_date=datetime.date(2016, 1, 1)
+    )
+
+    assert award.awarding_agency == agency1
+    assert award.period_of_performance_current_end_date == datetime.date(2016, 1, 1)
+    assert award.description == 'original award'
+
+    # adding an newer transaction with different info updates award's total
+    # obligation amt and also overrides other values
+    mommy.make(
+        'awards.Transaction',
+        id=999,
+        award=award,
+        awarding_agency=agency2,
+        period_of_performance_current_end_date=datetime.date(2010, 1, 1),
+        description='new description',
+        action_date=datetime.date(2017, 1, 1)
+    )
+
+    assert award.awarding_agency == agency2
+    assert award.period_of_performance_current_end_date == datetime.date(2010, 1, 1)
+    assert award.description == 'new description'
+
+
+@pytest.mark.django_db
+def test_award_update_from_earliest_transaction():
+    """Test awards fields that should be updated with most earliest transaction info."""
+
+    award = mommy.make('awards.Award')
+    mommy.make(
+        'awards.Transaction',
+        award=award,
+        # since this is the award's first transaction,
+        # the txn action_date will become the award
+        # signed date
+        action_date=datetime.date(2016, 1, 1)
+    )
+
+    # adding later transaction should not change award values
+    mommy.make(
+        'awards.Transaction',
+        award=award,
+        action_date=datetime.date(2017, 1, 1)
+    )
+    assert award.date_signed == datetime.date(2016, 1, 1)
+
+    # adding earlier transaction should update award values
+    mommy.make(
+        'awards.Transaction',
+        award=award,
+        action_date=datetime.date(2010, 1, 1)
+    )
+    assert award.date_signed == datetime.date(2010, 1, 1)
+
+
+@pytest.mark.django_db
+def test_award_update_obligated_amt():
+    """Test that the award obligated amt updates as child transactions change."""
+
+    award = mommy.make('awards.Award', total_obligation=1000)
+    mommy.make(
+        'awards.Transaction',
+        award=award,
+        federal_action_obligation=1000,
+        _quantity=5
+    )
+    assert award.total_obligation == 5000
+
+
+@pytest.mark.django_db
+def test_award_update_from_contract_transaction():
+    """Test award updates specific to contract transactions."""
+
+    # for contract type transactions,
+    # the potential_total_value_of_award field
+    # should updte the corresponding field on the award table
+    award = mommy.make('awards.Award')
+    txn = mommy.make('awards.Transaction', award=award)
+    mommy.make(
+        'awards.TransactionContract',
+        transaction=txn,
+        potential_total_value_of_award=1000
+    )
+    assert award.potential_total_value_of_award == 1000
+
+
```

Comment:
Marking this with `@pytest.mark.skip` might be better than an empty test - it will provide a reminder that we've got a desired but unwritten test